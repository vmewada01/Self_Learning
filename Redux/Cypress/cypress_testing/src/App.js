import logo from './logo.svg';
import './App.css';
import { useState } from 'react';
import  axios from "axios";
import Form from './components/Form';


function App() {
 const [counter ,setCounter] = useState(0)

  return (
    <div className="App">
     <h1>The Count is :  {counter}</h1>
     <button className='addCounter' onClick={()=> setCounter((prev)=> prev+1 )}>Add</button>
     <button className='reduceCounter' onClick={()=> setCounter((prev)=> prev-1)}>Reduce</button>
     <Form />
    </div>
  );
}

export default App;
