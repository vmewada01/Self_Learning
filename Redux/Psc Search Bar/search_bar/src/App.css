.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

input {
  padding: 1rem;
  width: 80%;
  border: 1px solid black;
  border-radius: 1rem;
  font-size: 1rem;
}

.search {
  border: 1px solid black;
  display: flex;
  margin: auto;
  padding-left: 1rem;
  padding: 1rem;
  width: 80%;
 cursor: pointer;
  
}
.search:hover{
  background-color: rgba(0,0,0,0.08);
  color: black;
}

.searchDiv {
  height: 250px;
  overflow: auto;
  margin: auto;
}
