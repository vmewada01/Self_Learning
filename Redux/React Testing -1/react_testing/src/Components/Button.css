.button {
    margin: 1rem;
    padding: 1rem;
    font-size: 1rem;
    border-radius: 0.5rem;
    cursor: pointer;
    
}
.red{
    background-color: red;
    color: black;
}

.blue{
    background-color: blue;
    color: black;
}
.orange{
    background-color: orange;
    color: black;
}
.ghost{
  border: none;
}
.outline{
 border: 1px solid green;
 color: black;
}
.green{
    background-color: green;
}