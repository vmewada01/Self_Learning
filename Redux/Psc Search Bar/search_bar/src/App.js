import logo from './logo.svg';
import './App.css';
import SeacrhBar from './components/SeacrhBar';
import { useCallback, useEffect, useState } from 'react';
import Countries from './Utils/searchData';

function App() {
  const [textValue, setTextValue]= useState("")
  const [query, setQuery] = useState([])
 
  const queryHandler= useCallback((value)=> {
       setTextValue(value)
  },[])


    useEffect(()=> {
    if(textValue===""){
      setQuery([])
    }
    else{
      let newCountries = Countries.filter((item)=> {
        return item.country.toLowerCase().indexOf(textValue)
        !== -1 ? true: false
      })
      .map((item)=> item.country);
      setQuery(newCountries)
    }
    },[textValue])

  return (
    <div className="App">
      <h1>Search Bar PSC</h1>
      <h3>The search Query "{textValue}"</h3>
      <SeacrhBar query={query} queryHandler={queryHandler} />
    </div>
  );
}

export default App;
