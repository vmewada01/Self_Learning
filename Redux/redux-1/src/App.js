import logo from "./logo.svg";
import "./App.css";
import Counter from "./Components/Counter";
import Todo from "./Todo/Todo";
import Todo_Id from "./Todo/Todo_Id";
import { useSelector } from "react-redux";


function App() {
  const isAuth = useSelector((store) => store.AuthReducer.isAuth);
 
  return (
    <div className="App">
      <Counter />
     <Todo_Id />
     {isAuth &&  <Todo />}
    </div>
  );
}

export default App;
