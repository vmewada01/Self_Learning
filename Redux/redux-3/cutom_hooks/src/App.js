import logo from "./logo.svg";
import "./App.css";
import useTimeout from "./UseHooks/useTimeout";
import Checkout from "./Components/Checkout";
import useFetch from "./UseHooks/useFetch";
import { useEffect, useState } from "react";
import useDebounce from "./UseHooks/useDebounce";
import useThrottle from "./UseHooks/useThrottle";
import useTimer from "./UseHooks/useTimer";

function App() {
  // const data = useTimeout();
  // const { item, error, loading } = useFetch("http://api.github.com/users?q=Masai");
  // console.log(item, error, loading);

  // const [count, setCount] = useState(0);

  // const handleScrollEvent = () => {
  //   setCount((prev) => prev + 1);
  // };

  // useThrottle(() => {}, 3000);

  // // useDebounce(() => {}, 2000);
  // useEffect(() => {
  //   window.addEventListener("scroll", handleScrollEvent);
  //   return () => {
  //     window.removeEventListener("scroll", handleScrollEvent);
  //   };
  // }, []);

  const {timer,handleStop, handleStart, handleReset} = useTimer()

  const formatTime = (milliseconds) => {
    const hours = Math.floor(milliseconds / (1000 * 60 * 60));
    const minutes = Math.floor((milliseconds % (1000 * 60 * 60)) / (1000 * 60));
    const seconds = Math.floor((milliseconds % (1000 * 60)) / 1000);
    const remainingMilliseconds = milliseconds % 1000;

    const formattedHours = String(hours).padStart(2, "0");
    const formattedMinutes = String(minutes).padStart(2, "0");
    const formattedSeconds = String(seconds).padStart(2, "0");
    const formattedMilliseconds = String(remainingMilliseconds).padStart(
      3,
      "0"
    );

    return `${formattedHours}:${formattedMinutes}:${formattedSeconds}.${formattedMilliseconds}`;
  };



  return (
    <div className="App">
      {/* <div style={{ position: "sticky", top: "100px" }}>
        <h3>scroll Count : {count}</h3>
      </div>
      <h1>Components</h1>
      <h2> {data ? "I m Ready " : "I m confused"} </h2>
      {/* <Checkout/> */}
      {/* <div>{loading ? "Complete" : "LOADING....!!!"}</div>
      <div>{error ? "Everything looks fine..!!" : "Something went wrong"}</div>
      {item.map((abc) => {
        return <div key={abc.id}>{abc.login}</div>;
      })} */}
 
          <h1>`{formatTime(timer)}`</h1>
        <div>
        <button onClick={handleStart}>START</button>
        <button onClick={handleStop}>STOP</button>
        <button onClick={handleReset}>RESET</button>
      
        </div>
       

    </div>
  );
}

export default App;
