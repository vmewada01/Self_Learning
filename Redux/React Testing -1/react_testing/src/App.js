import logo from "./logo.svg";
import "./App.css";
import { useState } from "react";
import Button from "./Components/Button";

function App() {

  const [theme, setTheme] = useState("light")
  return <div className="App">
 <h2>Theme is {theme}</h2>
 <div> Click Me </div>
   <Button
 
    onClick={()=> setTheme(theme==="light" ? "dark": "light")}
    colorscheme={"red"}
    variant={"outline"}
    >
    Click Me
    </Button> 
  </div>;
}

export default App;
