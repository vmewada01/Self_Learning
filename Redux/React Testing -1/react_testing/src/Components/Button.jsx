import React from "react";
import "./Button.css"
const Button = ({children, colorscheme, variant,onClick}) => {
  return <button
  data-testId="customButton"
  className={`button ${colorscheme || "blue"} ${variant || "ghost"}`}
  onClick={onClick}  
  >{children}</button>;
};

export default Button;
