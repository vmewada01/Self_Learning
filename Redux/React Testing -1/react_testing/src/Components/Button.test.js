import App from "../App";
import Button from "./Button";

import userEvent from "@testing-library/user-event";
import { render, screen, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer"


describe("Testing the custom Button", () => {
  it("Should be present in the DOM", () => {
    render(<Button>Click Me</Button>);
    //screen.debug()
    let button = screen.getByText("Click Me");
    expect(button).toBeInTheDocument();
  });

  it("Should have a button with text Click Me", () => {
    render(<App></App>);
    let button = screen.getByTestId("customButton");
    expect(button).toBeInTheDocument();
  });

  it("Should have render without Children", () => {
    render(<Button></Button>);
    let button = screen.getByTestId("customButton");
    expect(button).toBeEmptyDOMElement();
  });

//   it("Should  toggle the theme", () => {
//     render(<App></App>);
//     let element = screen.getByText("Theme is light");
//     expect(element).toHaveTextContent("light");

//     let button = screen.getByTestId("customButton");
//        userEvent.click(button);

//     expect(element).toHaveTextContent("dark");
//     // userEvent.click(button);

//     // expect(element).toHaveTextContent("light");
//   });


  it("Should toggle the theme", () => {
    render(<App></App>);
    let element = screen.getByText("Theme is light");
    expect(element).toHaveTextContent("light");

    let button = screen.getByTestId("customButton");
    userEvent.click(button);

    expect(element).toHaveTextContent("dark");
    userEvent.click(button);
    expect(element).toHaveTextContent("light");
  });

  it("Should call the given function", () => {
    const mockFunc = jest.fn(); // () => {}

    render(<Button onClick={mockFunc}>Click Me</Button>);

    const button = screen.getByTestId("customButton");
    fireEvent.click(button);
    fireEvent.click(button);

    // expect(mockFunc).toBeCalled();
    expect(mockFunc).toHaveBeenCalledTimes(2);
    // toBeCalledTimes(2);
  });

  //Snapshot ->

  it("Should match the snapshot", () => {
    const tree = renderer
      .create(
        <Button colorScheme={"green"} variant={"outline"}>
          Increment Counter
        </Button>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });





});
