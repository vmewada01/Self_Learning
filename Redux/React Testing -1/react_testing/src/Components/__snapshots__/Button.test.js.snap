// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the custom Button Should match the snapshot 1`] = `
<button
  className="button blue outline"
  data-testId="customButton"
>
  Increment Counter
</button>
`;
